resteasy-core/src/main/java/org/jboss/resteasy/core/ConstructorInjectorImpl.java

```diff
@@ -11,6 +11,8 @@
 import org.jboss.resteasy.spi.ValueInjector;
 import org.jboss.resteasy.spi.metadata.ConstructorParameter;
 import org.jboss.resteasy.spi.metadata.ResourceConstructor;
+import org.jboss.resteasy.spi.BadRequestException;
+import org.jboss.resteasy.spi.NotFoundException;

```

Comment:
Similar comments to the other below on the exception imports (sorry, noticed them before this)

Reply:
Yes org.jboss.resteasy.spi.NotFoundException use is intentional.  I created it.  Just like 
org.jboss.resteasy.spi.BadRequestException which is pre-existing and used in several places and
by design carries the http error code in it.  I created org.jboss.resteasy.spi.NotFoundException to do the same.

I don't see that  org.jboss.resteasy.spi.BadRequestException or org.jboss.resteasy.spi.NotFoundException is deprecated?

Was it your intention to replace all uses of org.jboss.resteasy.spi.BadRequestException with
javax.ws.rs.BadRequestException ..etc.   That would require some adjustments in resteasy's
response and error messaging code.

Reply:
@rsearls well, the class was deprecated by Bill Burke in May 2013 (https://github.com/resteasy/Resteasy/commit/6e549642d1aa8b9d9f4f25dc8574c7a7a8b1644a#diff-192636fff108ec83395dee4ce70b425fR10), when he basically made changes to use the corresponding exception from jaxrs spec (I believe that was because previously there was no such exception in old JAX-RS api version). The deprecated exception was removed from master years after (I think I did that when started the cleanup on master).
So your addition would basically bring back something that was previously deprecated and removed; the question is really whether that's actually needed or we can go with the JAX-RS standard annotation...

Reply:
Alessio and I have discussed this.  We agreed to accept this as is.  I will create a new jira
to replace this exception and BadRequestException with the jax-rs classes and make needed
adjustments in the resteasy